subsys/net/ip/net_context.c

```diff
@@ -2043,6 +2045,8 @@ static int recv_raw(struct net_context *context,
 		return ret;
 	}
 
+	net_context_set_state(context, NET_CONTEXT_LISTENING);
```

Comment:
Does it mean that all data before first `recv()` call will be dropped? If yes and we assume a case that someone wants to:
1) create a AF_PACKET socket
2) send request
3) receive reply

then we there is race condition here...

Reply:
I realized after sending this last night that there needs to be some rework in this part. I will revisit this in next version.